### PR TITLE
Add support for more network plugins, e.g. flannel and calico

### PR DESCRIPTION
**Pull request by feiskyer**
This PR makes CNI supports old-versioned plugins. And also adds support for more network plugins, e.g. flannel and calico

Closes: #69 #150

### CODE REVIEW

**Comment by feiskyer** [comment]
cc/ @resouer

File: docs/networking.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+{
+    "cniVersion": "0.3.1",
+    "type": "loopback"
+}
+EOF'
+```
+
+## [Flannel](https://github.com/coreos/flannel)
+
+Remove othe cni network configure if they are already configured:
```
Comment (resouer) [inline docs/networking.md]: typo: other

File: docs/networking.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+Then setup flannel plugin by running:
+
+```sh
+kubectl create -f https://github.com/coreos/flannel/raw/master/Documentation/kube-flannel-rbac.yml
+kubectl create -f https://github.com/coreos/flannel/raw/master/Documentation/kube-flannel.yml
+```
+
+## [Calico](https://www.projectcalico.org)
+
+Remove othe cni network configure if they are already configured:
```
Comment (resouer) [inline docs/networking.md]: typo: other

File: docs/networking.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+```sh
+rm -f /etc/cni/net.d/*
+```
+
+Then setup calico plugin by running:
+
+```sh
+kubectl apply -f https://docs.projectcalico.org/v2.4/getting-started/kubernetes/installation/hosted/kubeadm/1.6/calico.yaml
```
Comment (resouer) [inline docs/networking.md]: Is this`kubeadm/1.6/` implying we should we use k8s 1.6?

File: docs/networking.md (hunk 1 above)
Comment (feiskyer) [inline docs/networking.md]: ack

File: docs/networking.md (hunk 2 above)
Comment (feiskyer) [inline docs/networking.md]: ack

File: docs/networking.md (hunk 3 above)
Comment (feiskyer) [inline docs/networking.md]: No, it origins from calico's [documentation](https://docs.projectcalico.org/v2.4/getting-started/kubernetes/installation/hosted/kubeadm/), and supports kubernetes v1.6+

**Pull request merged**